src/vs/workbench/contrib/remote/browser/remoteStartEntry.ts

```diff
@@ -41,86 +27,35 @@ type RemoteStartActionEvent = {
+	private readonly remoteExtensionId!: string;
+	private startCommand!: string;
 
 	constructor(
-		@IQuickInputService private readonly quickInputService: IQuickInputService,
 		@ICommandService private readonly commandService: ICommandService,
 		@IProductService private readonly productService: IProductService,
-		@IExtensionService private readonly extensionService: IExtensionService,
 		@IExtensionManagementService private readonly extensionManagementService: IExtensionManagementService,
 		@IWorkbenchExtensionEnablementService private readonly extensionEnablementService: IWorkbenchExtensionEnablementService,
-		@IExtensionGalleryService private readonly extensionGalleryService: IExtensionGalleryService,
 		@ITelemetryService private readonly telemetryService: ITelemetryService,
-		@IContextKeyService private readonly contextKeyService: IContextKeyService,
-		@IOpenerService private readonly openerService: IOpenerService) {
+		@IContextKeyService private readonly contextKeyService: IContextKeyService) {
 
 		super();
-		const enable = this.extensionGalleryService.isEnabled() && (this.productService.remoteExtensionTips ?? false) && this.productService.quality !== 'stable';
-		registerConfiguration(enable);
-
-		if (isWeb) {
-			const remoteExtensionTips = this.productService.remoteExtensionTips?.['tunnel'];
-			this.remoteExtensionMetadata = remoteExtensionTips ? [{
-				id: remoteExtensionTips.extensionId,
-				installed: false,
-				friendlyName: remoteExtensionTips.friendlyName,
-				remoteCommands: [],
-				isPlatformCompatible: false,
-				dependencies: [],
-				helpLink: remoteExtensionTips.startEntry?.helpLink ?? '',
-				startConnectLabel: remoteExtensionTips.startEntry?.startConnectLabel ?? '',
-				startCommand: remoteExtensionTips.startEntry?.startCommand ?? '',
-				priority: remoteExtensionTips.startEntry?.priority ?? 10
-			}] : [];
+		if (!isWeb) {
+			return;
 		}
-		else {
-			const remoteExtensionTips = { ...this.productService.remoteExtensionTips, ...this.productService.virtualWorkspaceExtensionTips };
-			this.remoteExtensionMetadata = Object.values(remoteExtensionTips).filter(value => value.startEntry !== undefined).map(value => {
-				return {
-					id: value.extensionId,
-					installed: false,
-					friendlyName: value.friendlyName,
-					remoteCommands: [],
-					isPlatformCompatible: false,
-					dependencies: [],
-					helpLink: value.startEntry?.helpLink ?? '',
-					startConnectLabel: value.startEntry?.startConnectLabel ?? '',
-					startCommand: value.startEntry?.startCommand ?? '',
-					priority: value.startEntry?.priority ?? 10,
-					supportedPlatforms: value.supportedPlatforms
-				};
-			});
 
-			this.remoteExtensionMetadata.sort((ext1, ext2) => ext1.priority - ext2.priority);
+		const remoteExtensionTips = this.productService.remoteExtensionTips?.['tunnel'];
+		if (!remoteExtensionTips) {
+			return;
 		}
+		this.startCommand = remoteExtensionTips.startEntry?.startCommand ?? '';
```

Comment:
With the fallback to a default value here, the non-null assertions above should not be necessary

Reply:
If I remove the non-null assertion, I see the `...has no initializer and is definitely not initialized in the constructor`  compiler error because of the `if` conditions I have before assignment here. 

What I ideally want to do is only have this remoteStartEntry be a workbench contribution for the web.

Reply:
Thanks for the pointers. Moved remoteStartEntry to be a web only contribution and removed the checks.